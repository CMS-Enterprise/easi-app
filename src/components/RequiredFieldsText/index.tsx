import React from 'react';
import { Trans } from 'react-i18next';
import classNames from 'classnames';

import RequiredAsterisk from 'components/shared/RequiredAsterisk';

const RequiredFieldsText = ({ className }: { className?: string }) => {
  return (
    <p className={classNames('text-base', className)}>
      <Trans
        i18nKey="action:fieldsMarkedRequired"
        components={{ asterisk: <RequiredAsterisk /> }}
      />
    </p>
  );
};

export default RequiredFieldsText;
