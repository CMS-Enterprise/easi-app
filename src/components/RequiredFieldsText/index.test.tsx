import React from 'react';
import { render } from '@testing-library/react';

import RequiredFieldsText from '.';

describe('Required fields text component', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(<RequiredFieldsText />);

    expect(asFragment()).toMatchSnapshot();
  });
});
