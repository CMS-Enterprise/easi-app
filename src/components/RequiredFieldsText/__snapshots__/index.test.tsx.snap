// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Required fields text component > matches the snapshot 1`] = `
<DocumentFragment>
  <p
    class="text-base"
  >
    Fields marked with an asterisk (
    <span
      class="text-red margin-x-05"
    >
      *
    </span>
    ) are required.
  </p>
</DocumentFragment>
`;
